### ISSUE TITLE
Web: Implement Prediction Page with ML Model Integration

### ISSUE THREAD

**Issue by jpequegn** [jpequegn/f1-predict#52]
## Overview
Implement the full prediction page in the Streamlit web interface, integrating with trained ML models to generate real F1 race predictions with confidence scores and explanations.

## Context
Issue #15 established the web interface foundation with page stubs. This issue focuses on implementing the prediction page functionality using the ML models from Issue #9.

### 1. Race Selection Interface
- [ ] Load upcoming races from data/schedules
- [ ] Display race information (name, circuit, date, location)
- [ ] Support manual race selection from dropdown
- [ ] Show circuit details and characteristics

### 2. Model Selection & Configuration
- [ ] Model selector (Ensemble, XGBoost, LightGBM, Random Forest)
- [ ] Display model metadata (accuracy, training date, version)
- [ ] Advanced options (expandable):
  - Weather condition override
  - Temperature settings
  - Tire strategy selection
  - Safety car probability

### 3. Prediction Generation
- [ ] Integrate with trained ML models from src/f1_predict/models/
- [ ] Load feature data for selected race
- [ ] Generate predictions with confidence scores
- [ ] Handle prediction errors gracefully
- [ ] Show loading spinner during generation

### 4. Results Display

**Predicted Podium:**
- [ ] Display top 3 positions with driver names
- [ ] Show confidence percentages
- [ ] Team colors/logos

**Full Race Order:**
- [ ] Interactive table with all positions
- [ ] Driver, team, confidence columns
- [ ] Sortable by confidence
- [ ] Progress bar for confidence visualization

### 5. Prediction Explanations
- [ ] Feature importance chart (using SHAP values if available)
- [ ] Top contributing factors
- [ ] Model reasoning explanation
- [ ] Collapsible explanation section

### 6. Confidence Visualization
- [ ] Chart showing confidence distribution across positions
- [ ] Uncertainty indicators
- [ ] Comparison with historical accuracy

### 7. Export & Sharing
- [ ] Download prediction as CSV
- [ ] Download prediction as JSON
- [ ] Copy to clipboard
- [ ] Share link (if implementing persistence)

### Integration Points
```python
# Load model
from f1_predict.models import load_model

model = load_model("ensemble")  # or xgboost, lightgbm, random_forest

# Generate prediction
prediction = model.predict(race_features)

# Get feature importance
importance = model.get_feature_importance()
```

### Data Flow
1. User selects race → Load race data
2. User selects model → Load trained model
3. User clicks predict → Generate features → Run model → Display results
4. User explores → Show explanations, charts, export options

### State Management
```python
import streamlit as st

# Cache model loading
@st.cache_resource
def load_prediction_model(model_type):
    return load_model(model_type)

# Cache race data
@st.cache_data(ttl=3600)
def load_race_data(race_id):
    return get_race_features(race_id)
```

### Performance
- Model loading: <2 seconds (cached)
- Prediction generation: <5 seconds
- Results rendering: <1 second

### Error Handling
- Missing model files → Clear error message + download instructions
- Invalid race data → Graceful fallback with explanation
- Prediction failures → Error display with retry button

### Mobile Responsiveness
- Stacked layout on mobile (<768px)
- Touch-friendly controls
- Scrollable results table

## Acceptance Criteria

- [ ] Users can select any upcoming race
- [ ] Users can choose from 4 ML models
- [ ] Predictions generate within 5 seconds
- [ ] Results display predicted podium with confidence
- [ ] Full race order shown in interactive table
- [ ] Feature importance visualization available
- [ ] Export to CSV/JSON works correctly
- [ ] Mobile layout functions properly
- [ ] Error handling is comprehensive
- [ ] Unit tests pass with 80%+ coverage

## Dependencies

- **Issue #9**: Trained ML models (ensemble, xgboost, lightgbm, random_forest)
- **Issue #15**: Web interface foundation (completed)
- Feature engineering pipeline for generating race features

## Estimated Effort

**Complexity**: Medium  
**Duration**: 3-4 days  
**Priority**: High
